### ISSUE TITLE
feat(router): API for determining if a route is active

### ISSUE THREAD

**Issue by matsko** [angular/angular#3209]
I've come across a challenge when using the new router to build out a navigation menu of tabs. Each tab contains a link to a route where a different component takes over:

```html
<nav>
  <a [router-link]="['./home']">Home</a>
  <a [router-link]="['./about']">About</a>
  <a [router-link]="['./forum']">Forum</a>
</nav>

@RouteConfig([
  new Route({ path:"...", as: 'home' })
])
```

I want to add an `active` CSS class to the corresponding tab that is selected. In order to do this I've had to create a component method on the root component which does something like this:

```js
class RootCmp {
   constructor(public location: Location) { }
   isRoute(alias):boolean {
      var path = this.location.path().split('/')[0];
      path = path[0] == '/' ? path.substr(1) : path;
      switch(alias) {
         case 'home': return path === 'home';
         case 'about': return path === 'about';
         case 'forum': return path === 'forum';
      }
   }
}
```

This gets even more complex when the URL paths grow.

Seeing as the `as` value is already attached to the route, could we somehow have that be relayed to the router?

Something like:

```js
class RootCmp {
   constructor(public router: Router) { }
   isRoute(alias):boolean {
      return router.isActive(alias);
   }
}
```

For child routes we could implement a reverse list of routes going from current router up to the root. So something like this:

```js
// this would work
router.isActive(['forum_post']);

//but then would also work.
router.isActive(['forum_post', 'forum']);
```

We could also maybe reuse the router-link builder syntax? How does this sound?

**Comment by matsko** [comment]
@btford @IgorMinar ping.

**Comment by matsko** [comment]
Maybe once the AST sugar syntax is ready we could have something like that?

**Comment by btford** [comment]
:+1: This is a feature we want.

**Comment by ludohenin** [comment]
I've tried something using directives [here](https://github.com/AngularShowcase/angular2-sample-app/blob/master/app/directives/set-active.ts).
It has also leaded me to another issue (not sure if I should create a new issue for that) describe [here](https://github.com/AngularShowcase/angular2-sample-app/blob/master/app/directives/set-active.ts#L31-L34). Basically it's not possible to bind up-front to `href` attribute generated by the router.

**Comment by btford** [comment]
Tracks internal bug: b/22617807

**Comment by Fincodr** [comment]
Please note that the route in the browser url is not necessarily the route that is active since the active route could be temporally changed (with custom outlet for example, to block certain routes when user is not logged in, but still showing the original url where the user will go after logging in).

Currently in my app I am using private variable router._currentInstruction.current.component to see which route is active but we need something better (and public) and it should reflect the actual component url we are currently seeing, not necessary what is in the url. Hope this helps :)

**Comment by Krever** [comment]
@btford 
Hi, is it possible to apply custom class (.active for example) instead of 'router-link-active' ?
